mithril-common/src/crypto_helper/genesis.rs

```diff
@@ -63,13 +67,25 @@ impl ProtocolGenesisSigner {
         self.secret_key.sign(message).into()
     }
 
-    /// Export the secret key from the genesis verifier to a file. TEST ONLY
-    #[doc(hidden)]
-    pub fn export_to_file(&self, secret_key_path: &Path) -> StdResult<()> {
-        let mut genesis_secret_key_file = File::create(secret_key_path)?;
-        genesis_secret_key_file.write_all(self.secret_key.to_json_hex().unwrap().as_bytes())?;
+    /// Export the keypair from the genesis verifier to files
+    pub fn export_keypair_to_files(&self, keypair_path: &Path) -> StdResult<(PathBuf, PathBuf)> {
+        let genesis_secret_key_path = keypair_path.join("genesis.sk");
+        {
+            let genesis_secret_key_payload = self.secret_key.to_json_hex().unwrap();
+            let mut genesis_secret_key_file = File::create(&genesis_secret_key_path)?;
+            genesis_secret_key_file.write_all(genesis_secret_key_payload.as_bytes())?;
+        }
+
+        let genesis_verification_key_path = keypair_path.join("genesis.vk");
+        {
+            let genesis_verification_key: ProtocolGenesisVerificationKey =
+                self.secret_key.verifying_key().into();
+            let genesis_verification_key_payload = genesis_verification_key.to_json_hex().unwrap();
```

Comment:
Isn't just a technical stuff to write the ProtocolKey into a file ? 
We can create a function for this so that what varies between the two keys is more visible.

This function may also build and return the PathBuf from keypair_path and file name.
 ```
   pub fn export_keypair_to_files(&self, keypair_path: &Path) -> StdResult<(PathBuf, PathBuf)> {
        let genesis_secret_key_path = keypair_path.join("genesis.sk");
        self.write_key_to_file(&self.secret_key, genesis_secret_key_path.clone())?;

        let genesis_verification_key_path = keypair_path.join("genesis.vk");
        self.write_key_to_file(
            &self.secret_key.verifying_key().into(),
            genesis_verification_key_path.clone(),
        )?;

        Ok((genesis_secret_key_path, genesis_verification_key_path))
    }

    fn write_key_to_file<T>(
        &self,
        key: &super::ProtocolKey<T>,
        key_path: PathBuf,
    ) -> Result<(), anyhow::Error>
    where
        T: Serialize + DeserializeOwned,
    {
        let key_payload = key.to_json_hex().unwrap();
        let mut key_file = File::create(&key_path)?;
        key_file.write_all(key_payload.as_bytes())?;
        Ok(())
    }
}
```

Reply:
I implemented the write to file in the `ProtocolKey` in https://github.com/input-output-hk/mithril/pull/2106/commits/280fc73252f1d3a5d04f7cab5e2ab7ef4ac2b128 and refactored the genesis tools to make use of it in https://github.com/input-output-hk/mithril/pull/2106/commits/4f092c57c9b7400e4d774ca814bda2c6b2f79393 :+1: